src/projection-multi/projection-multi-haskell-stack.el

```diff
@@ -0,0 +1,327 @@
+  `(,(concat project-type ":" target)
+    :command
+    ,(projection-haskell-stack--command "build" target "--run-benchmarks")
+    :annotation
+    ,(if documentation
+         (concat documentation ": Build and run")
+       (concat "Run benchmark"))))
+
+(defun projection-multi-haskell-stack--target-run-target
+    (target documentation project-type)
+  "`compile-multi' target generator function to run an executable Haskell Stack
+TARGET.
+The generated target will be prefixed with PROJECT-TYPE.  When set, the
+generated target will be annotated with DOCUMENTATION."
+  `(,(concat project-type ":" target ":run")
+    :command
+    ,(projection-haskell-stack--command "run" target)
+    :annotation
+    ,(if documentation
+         (concat documentation ": Run")
+       (concat "Run"))))
+
+
+
+;;;###autoload
+(defun projection-multi-haskell-stack-targets (&optional project-type)
+  "`compile-multi' target generator function for Haskell Stack projects.
+When set the generated targets will be prefixed with PROJECT-TYPE."
+  (setq project-type (or project-type "haskell-stack"))
+
+  (append
+   ;; Packages
+   (cl-loop for (package . documentation) in
+            (projection-multi-haskell-stack--packages)
+            append `(,(projection-multi-haskell-stack--package-build-target
+                       package documentation project-type)
+                     ,(projection-multi-haskell-stack--package-test-target
+                       package documentation project-type)
+                     ,(projection-multi-haskell-stack--package-bench-target
+                       package documentation project-type)
+                     ,(projection-multi-haskell-stack--package-haddock-target
+                       package documentation project-type)
+                     ,(projection-multi-haskell-stack--package-ghci-target
+                       package documentation project-type)
+                     ,(projection-multi-haskell-stack--package-clean-target
+                       package documentation project-type)))
+   ;; Targets
+   (cl-loop for (target . documentation) in
+            (projection-multi-haskell-stack--targets)
+            append
+            `(,@(cond
+                ;; Executable targets
+                ((projection-multi-haskell-stack--exe-p target)
+                 (list
+                  (projection-multi-haskell-stack--target-build-target
+                   target documentation project-type)
+                  (projection-multi-haskell-stack--target-run-target
+                   target documentation project-type)))
+                ;; Test suite targets
+                ((projection-multi-haskell-stack--test-p target)
+                 (list
+                  (projection-multi-haskell-stack--target-test-target
+                   target documentation project-type)))
+                ;; Benchmark targets
+                ((projection-multi-haskell-stack--bench-p target)
```

Comment:
This does a regex search for every target every time someone calls compile-multi. And this specific function is unfortunately not cached. Can you encode the target kind as a symbol in the cached generator functions so checking for it is constant?

```text
;; In cache
(propertize target 'compile-multi-haskell-stack-kind 'library)
;; In here
(get-text-property 0 'compile-multi-haskell-stack-kind target)
```